File: MPI/mpicheck/tests/functions/3.0/MPI_Compare_and_swap.f

        
        program main
        include 'mpif.h'
        
        TYPE(INTEGER) var_0(10)
       TYPE(INTEGER) var_1(10)
       TYPE(INTEGER) var_2(10)
       INTEGER var_3
       INTEGER var_4
       INTEGER(KIND=MPI_ADDRESS_KIND) var_5
       INTEGER var_6
       INTEGER var_7
        call mpi_compare_and_swap(var_0, var_1, var_2, var_3, var_4, var_5, var_6, var_7)
       call pmpi_compare_and_swap(var_0, var_1, var_2, var_3, var_4, var_5, var_6, var_7)
        end program main
    

File: MPI/mpicheck/tests/functions/3.0/MPI_Compare_and_swap.c
#include <mpi.h>
int main(char argc, char**argv)
{
    /* vars */
    const void *var_0;
    const void *var_1;
    void *var_2;
    MPI_Datatype var_3;
    int var_4;
    MPI_Aint var_5;
    MPI_Win var_6;
    int ret;
    /* calls */
    ret = MPI_Compare_and_swap(var_0, var_1, var_2, var_3, var_4, var_5, var_6);
    ret = PMPI_Compare_and_swap(var_0, var_1, var_2, var_3, var_4, var_5, var_6);
    return 0;
}
